drivers/ieee802154/ieee802154_dw1000.c

```diff
@@ -150,22 +129,7 @@ static const struct dwt_hi_cfg dw1000_0_config = {
 static struct dwt_context dwt_0_context = {
 	.dev_lock = Z_SEM_INITIALIZER(dwt_0_context.dev_lock, 1,  1),
 	.phy_sem = Z_SEM_INITIALIZER(dwt_0_context.phy_sem, 0,  1),
-	.rf_cfg = {
```

Comment:
That is not the way this (and any 15.4) device should be configured. This is config per device and should remain so. PHY attributes like channel, data rate, PRF, SHR code would be set by 15.4 subsystem. No need for this change, it makes the driver unnecessary complicated. First 15.4 should be proper separated from the network subsystem and UWB support and appropriate services added there.

Reply:
DW1000 does support multiple configurations and in the real use cases, the user does have the needs to change the channel configuration at runtime. I agree that adding the configuration table is the application level's scope and should not be put here. Is it reasonable to add a vendor specific 15.4 config type like `IEEE802154_CONFIG_CUSTOM` so that there is a way to change the channel configuration table at runtime?

Reply:
> there is a way to change the channel configuration table at runtime?

There is a mgmt API for 802.15.4 that can be used to change configurations at runtime. See `include/net/ieee802154_mgmt.h` for details.

Reply:
I've checked that. The API only supports changing channel, TX power,  address and PAN_ID. There is not an API supporting to configure other channel parameters, such as data rates, PRF and SHR code.

Reply:
> The API only supports changing channel, TX power, address and PAN_ID. There is not an API supporting to configure other channel parameters, such as data rates, PRF and SHR code.

Yes, I know. I should have been more clearer, what I meant is that there is a generic API for changing the configuration, someone just needs to implement the options you mentioned in that API.

Reply:
Can you guys review my current changes, still not good or better? 

@jukkar I think some parameters are specific to UWB, maybe they can be grouped together as a channel phy parameters? If you don't mind, I can add that mgmt APIs to ask you guys to review. I would like to have these functions so that I can use in my projects.

Reply:
I do not know why this driver get you attention, but I would really appreciate if you could stop adding any new features to it, especially if you are not familiar with this device. The new features does not help and are not necessary at this point of development. Just adding new things to the 15.4 subsystem does not help too. First the 15.4 should be made an independent subsystem, this would allow simpler, better and faster testing. Also one would usually use this device without 6lo. I have started to work on and used it to develop this driver, but it was not suitable for the upstream. I am still on it but I can not spend much time on it right now. Second step would be adding rudimental support for the UWB PHY (and it not just a channel number)...

Reply:
Sorry for the bothering. I have used the device for a couple years but not related with 6lo. Basically we have followed the design DecaWave had done before, which could not be well aligned to Zephyr's architecture. Thanks for the explanation! And I would take this PR off from zephyr development tree, just keep it in my private git. I appreciate what you have done on this device and would like to leverage it as much as possible for the future projects. Let me know if I can be helpful for something.  Thank you!